synapse/lib/stormlib/edgeinfo.py

```diff
@@ -0,0 +1,145 @@
+import datetime
+
+import synapse.exc as s_exc
+import synapse.common as s_common
+
+import synapse.lib.stormtypes as s_stormtypes
+
+edge_set_desc = '''
+Set a documentation string for an edge that has been added to the graph.
+
+Example:
+    edge.set seen "description for the seen edge verb"
+'''
+
+edge_get_desc = '''
+Get the documentation string for an edge verb.
+'''
+
+edge_del_desc = '''
+Delete the documentation entry for a given edge verb.
+
+Note: If the edge is re-added to the graph the entry will re-populate.
+'''
+
+edge_list_desc = '''
+List all edges that have been added to the graph with their associated documentation.
+'''
+
+stormcmds = [
+    {
+        'name': 'edge.set',
+        'descr': edge_set_desc,
+        'cmdargs': (
+            ('name', {'help': 'The name of the edge verb.'}),
+            ('info', {'help': 'The documentation string.'}),
+        ),
+        'storm': '''
+            $lib.edge.set($cmdopts.name, $cmdopts.info)
+            $lib.print('Set edge info: {name}', name=$cmdopts.name)
+        ''',
+    },
+    {
+        'name': 'edge.get',
+        'descr': edge_get_desc,
+        'cmdargs': (
+            ('name', {'help': 'The name of the edge verb.'}),
+        ),
+        'storm': '''
+            $edef = $lib.edge.get($cmdopts.name)
+            if $edef {
+                $lib.print('{name}: {info}', name=$cmdopts.name, info=$edef)
+            } else {
+                $lib.print('Edge entry not found: {name}', name=$cmdopts.name)
+            }
+        ''',
+    },
+    {
+        'name': 'edge.del',
+        'descr': edge_del_desc,
+        'cmdargs': (
+            ('name', {'help': 'The name of the edge verb.'}),
+        ),
+        'storm': '''
+            $lib.edge.del($cmdopts.name)
+            $lib.print('Deleted edge entry: {name}', name=$cmdopts.name)
+        ''',
+    },
+    {
+        'name': 'edge.list',
+        'descr': edge_list_desc,
+        'storm': '''
+            $elist = $lib.edge.list()
+            if $elist {
+                $lib.print('name       edited               info')
+                for ($name, $edef) in $elist {
+                    $name = $name.ljust(10)
+                    $edited = $edef.edited.ljust(20)
+                    $lib.print('{name} {edited} {info}', name=$name, edited=$edited, info=$edef.info)
+                }
+            } else {
+                $lib.print('No edge entries found')
+            }
+        ''',
+    },
+]
+
+class LibEdge(s_stormtypes.Lib):
+
+    def addLibFuncs(self):
+        self.locls.update({
+            'set': self._methEdgeSet,
+            'get': self._methEdgeGet,
+            'del': self._methEdgeDel,
+            'list': self._methEdgeList,
+        })
+
+    async def _methEdgeSet(self, name, info):
```

Comment:
We should probably make this either `setdoc()` or a `set()` method that takes a prop name which is limited for now to `doc` to have appropriate name abstraction for the functionality

Reply:
Updated with a hard coded list (for now) of supported extended props